### PR TITLE
highlights(zig): highlight enum,union,error and custom number type

### PR DESCRIPTION
**Pull request by maxxnino**

### CODE REVIEW

File: queries/zig/highlights.scm [hunk 1]
```diff
@@ -27,13 +27,19 @@
...
+(struct_construction
+  (type_identifier) @constructor
 )
-
 ;; other identifiers
 (type_identifier) @type
+(custom_number_type) @type.builtin
 (primitive_type) @type.builtin
 (field_identifier) @field
+(enum_identifier) @field
```
Comment (vigoux) [inline queries/zig/highlights.scm:40]: I think we have an `enum` type that should be used here.

**Comment by vigoux** [comment]
Hi and thanks for this first contribution !
I left a little comment on your changes, and started our CI to check the changes!

Welcome in the contributor community !

**Comment by maxxnino** [comment]
Thank you. I will update at the weekend. Btw, I maintain my own fork of tree-sitter-zig. Fix some of the existing problems too. I know, it's not perfect yet. https://github.com/Himujjal/tree-sitter-zig/issues/8

File: queries/zig/highlights.scm (hunk 1 above)
Comment (maxxnino) [reply queries/zig/highlights.scm:40]: The enum_identifier is the green box in the image below, i think it should be either **constant** or **field**

enum

**Review by theHamsta — APPROVED** [review]

File: queries/zig/highlights.scm (hunk 1 above)
Comment (vigoux) [reply queries/zig/highlights.scm:40]: I'd rather have that a constant, as this is not exactly a field... Sorry to b annoying :wink:

Comment (maxxnino) [reply queries/zig/highlights.scm:40]: I tried to find what is [highlight](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/lua/nvim-treesitter/highlight.lua) for enum type is, is it a `@type.enum`.

Comment (vigoux) [reply queries/zig/highlights.scm:40]: Then that's not the one we are searching for, indeed.

Comment (maxxnino) [reply queries/zig/highlights.scm:40]: > I think we have an `enum` type that should be used here.

I just want to know what is the `enum` type you mentioned above. For the sake of improve me knowledge.

Comment (vigoux) [reply queries/zig/highlights.scm:40]: It was just a mistake from me, I thought `enum` existed, which is not true :wink:

Comment (vigoux) [reply queries/zig/highlights.scm:40]: Could you make that a `constant` though ? As this is technically a constant ?

Comment (maxxnino) [reply queries/zig/highlights.scm:40]: I have a day off tomorrow. I will make it

**Review by vigoux — APPROVED** [review]

File: queries/zig/highlights.scm (hunk 1 above)
Comment (maxxnino) [reply queries/zig/highlights.scm]: It's a constant now, i have more highlight, should i make a separate pull request or add more commit to this pull request, or combine it in one commit

Comment (vigoux) [reply queries/zig/highlights.scm]: As you wish, we can make that go further already, and see for the rest afterwards ?

Comment (maxxnino) [reply queries/zig/highlights.scm]: Untitled - Frame 0

**Review by vigoux — APPROVED** [review]

File: queries/zig/highlights.scm (hunk 1 above)
Comment (vigoux) [reply queries/zig/highlights.scm]: Great !

**Pull request merged**